From 622055b7da6fd2e4a2c7884dfbc66921082547d2 Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Thu, 11 Apr 2019 18:12:17 +0300
Subject: [PATCH 152/386] arm64: dts: cn9131: add cn9131-db device tree with
 setup B

The cn913x DB board is a modular board,
which supports multiple possible configurations.

Topology for setup B:
AP-MPPs:
	- SDIO, UART
CP0-SerDes:
	- lane0: PCIe0
	- lane1: PCIe0
	- lane2: PCIe0
	- lane3: PCIe0
	- lane4: SFI0
	- lane5: SATA1
CP0-MPPs:
	- RGMII1, RGMII2, SDIO, I2C, NAND
CP1-SerDes:
	- lane0: PCIe0
	- lane1: PCIe0
	- lane2: SFI0
	- lane3: USB3_1
	- lane4: SGMII1
	- lane5: SATA1
CP1-MPPs:
	- I2C, SPI1

Change-Id: Icb7eefd43bbf577dca9188e0233d293494197791
Signed-off-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/7251
Reviewed-by: Grzegorz Jaszczyk <Grzegorz.Jaszczyk@cavium.com>
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 .../bindings/arm/marvell/cn913x.txt           |  5 ++-
 arch/arm64/boot/dts/marvell/Makefile          |  1 +
 arch/arm64/boot/dts/marvell/cn9131-db-B.dts   | 41 +++++++++++++++++++
 3 files changed, 46 insertions(+), 1 deletion(-)
 create mode 100644 arch/arm64/boot/dts/marvell/cn9131-db-B.dts

diff --git a/Documentation/devicetree/bindings/arm/marvell/cn913x.txt b/Documentation/devicetree/bindings/arm/marvell/cn913x.txt
index 15e9f2506cd7..d5ce92b4ef30 100644
--- a/Documentation/devicetree/bindings/arm/marvell/cn913x.txt
+++ b/Documentation/devicetree/bindings/arm/marvell/cn913x.txt
@@ -15,7 +15,6 @@ Boards:
       when the board being used is the OcteonTX2 CN9130 Development board with
       Setup A.
 
-Boards:
    - "marvell,cn9130-db-B"
       when the board being used is the OcteonTX2 CN9130 Development board with
       Setup B.
@@ -24,6 +23,10 @@ Boards:
       when the board being used is the OcteonTX2 CN9131 Development board with
       Setup A.
 
+   - "marvell,cn9131-db-B"
+      when the board being used is the OcteonTX2 CN9131 Development board with
+      Setup B.
+
    - "marvell,cn9132-db-A", "marvell,cn9132-db"
       when the board being used is the OcteonTX2 CN9132 Development board with
       Setup A.
diff --git a/arch/arm64/boot/dts/marvell/Makefile b/arch/arm64/boot/dts/marvell/Makefile
index 657f3b314e4e..2c582578cadf 100644
--- a/arch/arm64/boot/dts/marvell/Makefile
+++ b/arch/arm64/boot/dts/marvell/Makefile
@@ -27,4 +27,5 @@ dtb-$(CONFIG_ARCH_MVEBU) += armada-3900-db-vd-B.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += cn9130-db-A.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += cn9130-db-B.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += cn9131-db-A.dtb
+dtb-$(CONFIG_ARCH_MVEBU) += cn9131-db-B.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += cn9132-db-A.dtb
diff --git a/arch/arm64/boot/dts/marvell/cn9131-db-B.dts b/arch/arm64/boot/dts/marvell/cn9131-db-B.dts
new file mode 100644
index 000000000000..f1ffbfca034e
--- /dev/null
+++ b/arch/arm64/boot/dts/marvell/cn9131-db-B.dts
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2019 Marvell International Ltd.
+ *
+ * SPDX-License-Identifier:    GPL-2.0
+ * https://spdx.org/licenses
+ */
+
+#include "cn9130-db-B.dts"
+#include "cn9131-db.dtsi"
+
+/ {
+	model = "Marvell CN9131 development board (CP NAND) setup(B)";
+	compatible = "marvell,cn9131-db-B", "marvell,armada-ap807-quad",
+		     "marvell,armada-ap807";
+};
+
+/* RJ45 CON55 via 3310 */
+&cp1_eth0 {
+	status = "okay";
+	phy-mode = "10gbase-kr";
+	phy = <&phy8>;
+	phys = <&cp1_comphy2 0>;
+};
+
+/* CON50 */
+&cp1_eth1 {
+	status = "okay";
+	phy-mode = "sgmii";
+	/* Generic PHY, providing serdes lanes */
+	phys = <&cp1_comphy4 1>;
+	managed = "in-band-status";
+	sfp = <&cp1_sfp_eth1>;
+};
+
+&cp1_xmdio {
+	status = "okay";
+	phy8: ethernet-phy@8 {
+		compatible = "ethernet-phy-ieee802.3-c45";
+		reg = <8>;
+	};
+};
-- 
2.17.1

